### DISCUSSION TITLE
Create row in many-many relationship

### DISCUSSION THREAD

**Discussion by rmkaymusic** [Budibase/budibase#9329] [Help]
We're working with budibase to create a front end to our mysql database of freelancers and the video projects they've worked on.

I've hit a stumbling block in implementation. There is a many<->many relationship between Projects and Freelancers. While editing a project, at the point of adding a freelancer to that project I would like to be able to:
 - select from a list the freelancers currently attached to a project
 - have a button next to this list saying 'create new freelancer' which brings up a modal dialog to create a new freelancer with the correct field values already filled so that the freelancer - once created - is attached to the project.

The only/best way I can see to do this is to:
 - use the many/many relationship picker to do the select list of freelancers
 - for the 'create' button have it trigger an action to create a row in the table that links Projects to freelancers, insert the current project into that row
 - then open a modal with a new person using the id of the project-freelancer link table row we just created as a parameter
 - bind this parameter to the foreign key field in the new freelancer row so it is connected to the linking table
 - add an action to 'save' on the new freelancer modal that inserts the id of the new freelancer row into the linking table we created earlier. Then save!

Is there an easier way? This seems like quite hard work!

**Comment by melohagan** [comment]
There's a couple of examples here of how to do One -> Many, but the principle should be similar.

 - https://docs.budibase.com/docs/form-steps
 - https://docs.budibase.com/docs/chaining-actions#action-referencing
